### PR TITLE
Release v4.0.0

### LINKED ISSUE #856: 4.0.0 on RubyGems?
Sorry if I missed an existing issue about this, I tried looking around but couldn't find anything. Since Rails 5 came out yesterday, is there any chance of a doorkeeper release compatible with it?

Thanks for all the work you do on the gem, it really is appreciated ❤️

### PR DESCRIPTION
**Pull request by connorshea**
Not sure if I can add a tag via PR, but everything else should be good to go!

Fixes #856

cc: @tute

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ User-visible changes worth mentioning.
 
 ## master
 
+## 4.0.0
+
 - [#834] Fix AssetNotPrecompiled error with Sprockets 4
```
Comment (tute) [inline NEWS.md]: There are some changes worth mentioning between the latest release and today's master:

- [#847] Specify Null option to timestamps
- [#843] Revert "Fix validation error messages"
- [#847] Specify Null option to timestamps

Cam you please add them? Thanks!

**Comment by tute** [comment]
Looking good! We'll release this soon with your commit. :)

**Comment by connorshea** [comment]
@tute would you like me to squash after adding those?

File: NEWS.md (hunk 1 above)
Comment (connorshea) [inline NEWS.md]: I assume you didn't mean to duplicate #847?

Comment (tute) [inline NEWS.md]: OMGosh, you are right. Should go to sleep soon. :D

**Comment by tute** [comment]
Yep, squash is better! Thank you!

**Comment by tute** [comment]
Doorkeeper 4 is out: https://rubygems.org/gems/doorkeeper. 🎉

**Comment by connorshea** [comment]
\o/ thanks!

**Pull request merged**